### ISSUE TITLE
Error: WIS HTTP 530

### ISSUE THREAD

**Issue by profucius** [toverainc/willow-inference-server#160]
Hello, a few days ago I started getting this new error on my ESP BOX devices when trying to speak a command. The device recognizes the "Alexa" call, but then immediately says `WIS HTTP 630` on the screen. It does not process any audio.

Has something changed?

**Comment by hamishcunningham** [comment]
I saw someone else commenting that the Tovera server is down, so if you're using that one that could be the reason

**Comment by profucius** [comment]
Yes I think that may be what's happening. I tried visiting [https://infer.tovera.io/api/willow](url) and it says

`Error 1033 Ray ID: 8ebd5c2afa1853db • 2024-12-02 18:25:31 UTC
Argo Tunnel error
`

I wonder if @kristiankielhofner is aware of this outage

**Comment by oywino** [comment]
I have recently been getting the same error message, but I'm not aware that I'm using anything named a "Tovera Server" (?)
Please explain how I can check this/be sure (or eliminate this possibility).

**Comment by profucius** [comment]
> I have recently been getting the same error message, but I'm not aware that I'm using anything named a "Tovera Server" (?)
> Please explain how I can check this/be sure (or eliminate this possibility).

Tovera server is the Willow Inference Server (WIS) that the dev is hosting. If you are not hosting your own WIS with a GTX card to interpret voice data, then you're using the Tovera server by default. 

You can check which server you're using in your Willow Application Server (WAS) under settings.

**Comment by profucius** [comment]
> Do you know if we can expect a fix anytime soon?

My thought is no. If you look at the dev history on both WAS and WIS, they've both been quite nearly abandoned. I suspect the developer has moved onto other projects. They may not even use this project in their own home, otherwise they would have noticed it being down already. 

The good news is there are other options. I haven't had the time yet to invest into this yet. You can either install your own WIS server locally and replace the URL in your settings, which would fix your voice recognition. Or you can drop Willow altogether and switch to a built-in HASS voice solution like the Wyoming Protocol. I'm leaning toward the latter myself, although it is more time investment and possibly won't work with our current hardware (or it may). If you're only interested in getting your current system running again, then install your own WIS, but that would require an always-on PC with at least a GTX level graphics card. 

Just FYI from my research so far. I may post an update here later once I've made my choice and spent the time to move on it.

**Comment by nikito** [comment]
If you check discord, myself and a few others also run the HA solution with local Wyoming Protocol servers that are GPU accelerated, using the same hardware (ESP Box and ESP Box 3 that Willow also uses), so it is possible if you wish to do so. :)

**Comment by profucius** [comment]
> If you check discord

Which discord would that be? I don't see a link to it anywhere in the repo readme.

**Comment by nikito** [comment]
https://heywillow.io/get-involved/

**Comment by profucius** [comment]
Thanks, taking a look in there now. I'm a bit disoriented to discord, could you share a direct link to the discussion you mentioned, using Wyoming Protocol with the ESP Box? I'd really like to join into that discussion and/or read about how you all did it, so that I can try doing it too. Thank you

**Comment by profucius** [comment]
Thanks, I opened a thread over there for further discussion. For anyone else interested, [it's here](https://discord.com/channels/1161666824178503720/1313906059374497854).

I'll leave this ticket open for now, since the original issue is still ongoing, and I want the dev to have a chance to see if (if they're at all interested in fixing it, or perhaps this ticket can be a place that the dev officially tells us they're not going to fix it)

**Comment by profucius** [comment]
#161 Very sad news...

**Comment by JayPearlman** [comment]
A few weeks ago I started working on a replacement endpoint due to the 530 error. It's awful that Kristian is no longer with us as he created so many wonderful things. I'm sharing the link to my code in this thread in hopes that it can keep the usage of the Willow project active. My code to use a Cloudflare Worker endpoint instead of WIS: https://github.com/JayPearlman/WIS-CF-Worker
